gateway-sp-comms/src/management_switch.rs

```diff
@@ -590,3 +603,14 @@ mod tests {
         }
     }
 }
+
+#[usdt::provider(provider = "gateway_sp_comms")]
+mod probes {
+    fn recv_packet(
+        _source: &SocketAddr,
+        _port: &SwitchPort,
+        _data: u64, // TODO actually a `*const u8`, but that isn't allowed by usdt
```

Comment:
I believe a `&[u8]` would work fine here, but it's been a while since I tried it.

Reply:
`&[u8]` works, but it gets serialized as a JSON array of ints, which makes the dtrace probe more annoying to deal with I think? At least, you can't do `tracemem()` on it.

Reply:
I can definitely see raw pointers as being a nice type to accept. Would that be reasonable for us to add?

Reply:
If you mean a reasonable amount of work, then yes I think so. We currently accept the builtin D primitive types, so I think extending it to support pointers to those types as well would not be a big lift.

Converting a pointer to an integer as is done here "works", though, and that will actually still be done under the hood. So I think it would only serve as documentation (which is valuable!).

Reply:
I filed oxidecomputer/usdt#55; I'll merge this PR with the `u64` workaround for now.